### ISSUE TITLE
Add matchesPrivateKey and matchesPublicKey

### ISSUE THREAD

**Issue by dignifiedquire** [diasdavid/js-peer-id#23]
These are needed in js-libp2p-secio. Reference implementation in go: https://github.com/ipfs/go-libp2p-peer/blob/master/peer.go#L55-L67

**Comment by diasdavid** [comment]
👍

**Comment by diasdavid** [comment]
@dignifiedquire can you add these methods? Right now secio is not doing this verification, right?

**Comment by dignifiedquire** [comment]
Yes currently we are not doing this verification, as these methods are missing.
